R/visualize.R

```diff
@@ -0,0 +1,154 @@
+#' Fairness Accuracy Tradeoff Visualization
+#'
+#' @description
+#' This function specialize in comparing the Fairness vs Accuracy between learners through visualizations.
+#' From the visualization, users could see the tradeoff between fairness metrics and accuracy.
+#' Those insights could help the users to choose the optimum model from their model sets.
+#' And the fairness consistency of the learners by visualizing the tradeoff for different resampling iterations.
+#' It could take multiple type of inputs, like (`Prediction()`) (`BenchmarkResult()`) and (`ResampleResult()`).
+#'
+#'
+#' @param predcitions (`Prediction()`) (`BenchmarkResult()`) (`ResampleResult()`)\cr
+#' The binary class prediction object that will be evaluated.
+#' * If provided as (`Prediction()`). Then only one point will indicate the accuracy and fairness metrics for the current predictions.
+#' * If provided as (`ResampleResult()`). Then the plot will compare the accuracy and fairness metrics for the same model, but different rasampling iterations.
+#' * If provided as (`BenchmarkResult()`). Then the plot will compare the accuracy and fairness metrics for all models and all rasampling iterations.
+#'
+#' @param fairness_measure (`Measure()`)\cr
+#' The fairness measures that will evaluated on predictions.
+#'
+#' @param data_task (`TaskClassif()`)\cr
+#' The data task that contains the protected column, only required when the class of predcitions is (`Prediction()`)
+#'
+#' @export
+#'
+#' @examples
+#'
+#' # Setup the Fairness Measures and tasks
+#' data_task = tsk("adult_train")
+#' learner = lrn("classif.ranger", predict_type = "prob")
+#' learner$train(data_task)
+#' predictions = learner$predict(data_task)
+#' design = benchmark_grid(
+#'   tasks = tsk("adult_train"),
+#'   learners = lrns(c("classif.ranger", "classif.rpart"),
+#'                  predict_type = "prob", predict_sets = c("train", "test")),
+#'   resamplings = rsmps("cv", folds = 3)
+#' )
+#'
+#' bmr = benchmark(design)
+#' fairness_measure = msr("fairness.tpr")
+#' fairness_measures = msrs(c("fairness.tpr", "fairness.fnr"))
+#'
+#' fairness_accuracy_tradeoff(predictions, fairness_measure, data_task)
+#' fairness_accuracy_tradeoff(bmr, fairness_measure)
+#'
+fairness_accuracy_tradeoff <- function(predcitions, ...){
+  UseMethod("fairness_accuracy_tradeoff")
+}
+
+fairness_accuracy_tradeoff.Prediction <- function(predictions, fairness_measure, data_task){
+  data = data.table(accuracy = predictions$score(msr("classif.acc")),
+                    fairness = predictions$score(fairness_measure, data_task))
+  ggplot(data, aes(x = accuracy, y=fairness)) + geom_point()
+}
+
+fairness_accuracy_tradeoff.BenchmarkResult <- function(predictions, fairness_measure){
+  data = data.table(model = predictions$score(fairness_measure)[,6],
```

Comment:
accuracy should not be pre-defined. instead, also add a param `accuracy_measure` that defaults to acc.
This holds for all plots.

Reply:
Fixed that